### PR TITLE
Embedding.from_pretrained factory

### PR DESCRIPTION
**Pull request by asoltysik**
Solves #4801 
Docs and tests to be done after maintainers review the changes.
My issues:

1. `nn.Embedding.load_pretrained` currently has only two parameters: `embeddings` and `freeze`. Should I include more? 
2. `nn.Embedding` now has weight parameter that defaults to `None` as well as `num_embeddings` and `embedding_dim` - that's redundant.
3. Considering the above, maybe it would be better to create a subclass instead of a class method?

### CODE REVIEW

File: torch/nn/modules/sparse.py [hunk 1]
```diff
@@ -119,6 +124,14 @@ def __repr__(self):
         s += ')'
         return s.format(name=self.__class__.__name__, **self.__dict__)
 
+    @classmethod
+    def load_pretrained(cls, embeddings, freeze=True):
+        rows, cols = embeddings.shape
+        print(embeddings)
```
Comment (apaszke) [inline torch/nn/modules/sparse.py]: This shouldn't be here

File: torch/nn/modules/sparse.py [hunk 2]
```diff
@@ -119,6 +124,14 @@ def __repr__(self):
         s += ')'
         return s.format(name=self.__class__.__name__, **self.__dict__)
 
+    @classmethod
+    def load_pretrained(cls, embeddings, freeze=True):
+        rows, cols = embeddings.shape
```
Comment (apaszke) [inline torch/nn/modules/sparse.py]: Check that `embeddings.dim() == 2` and raise a nice error message saying that it was expected ot be 2D

File: torch/nn/modules/sparse.py [hunk 3]
```diff
@@ -119,6 +124,14 @@ def __repr__(self):
         s += ')'
         return s.format(name=self.__class__.__name__, **self.__dict__)
 
+    @classmethod
+    def load_pretrained(cls, embeddings, freeze=True):
```
Comment (apaszke) [inline torch/nn/modules/sparse.py]: nit: I'd call it `from_pretrained`, since it's not doing any e.g. loading from dist nor it loads the state into an existing object. It's an alternative constructor

File: torch/nn/modules/sparse.py [hunk 4]
```diff
@@ -75,7 +75,7 @@ class Embedding(Module):
 
     def __init__(self, num_embeddings, embedding_dim, padding_idx=None,
                  max_norm=None, norm_type=2, scale_grad_by_freq=False,
-                 sparse=False):
+                 sparse=False, weight=None):
```
Comment (apaszke) [inline torch/nn/modules/sparse.py]: Can you change the `weight` argument to `_weight` (such that it is for internal usage only)?

**Comment by asoltysik** [comment]
@apaszke Review adressed, docs and test added.

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Thanks @asoltysik!
